src/fluree/db/api/query.cljc

```diff
@@ -109,23 +109,28 @@
                                          {:subject query})
 
           {:keys [opts t]} query
-          db*      (if-let [policy-opts (perm/policy-opts (cond-> opts did (assoc :did did)))]
+          opts*    (util/parse-opts opts)
+          query*   (assoc query :opts opts*)
+          db*      (if-let [policy-opts (perm/policy-opts
+                                         (cond-> opts* did (assoc :did did)))]
                      (<? (perm/wrap-policy db policy-opts))
                      db)
           db**     (-> (if t
                          (<? (time-travel/as-of db* t))
                          db*)
                        (assoc-in [:policy :cache] (atom {})))
-          query*   (-> query
+          query**  (-> query*
                        (update :opts assoc :issuer did)
                        (update :opts dissoc :meta)
```

Comment:
This is just a stylistic thing, so however the spirit leads, but you can put all the `dissoc` keys in one update.